### PR TITLE
Add environmental variable to control path of media

### PR DESCRIPTION
**Pull request by epsilon-phase**
This was a request made on the matrix chat that will make it possible to release a snap version, that environmental variable being `MEDIA_UPLOAD_DIRECTORY`

### CODE REVIEW

File: src/routes/medias.rs [hunk 1]
```diff
@@ -72,7 +72,12 @@ pub fn upload(
                     .map(|ext| format!(".{}", ext))
             })
             .unwrap_or_default();
-        let dest = format!("static/media/{}{}", GUID::rand().to_string(), ext);
+        let dest = format!(
+            "{}/{}{}",
+            CONFIG.media_directory,
+            GUID::rand().to_string(),
```
Comment (igalic) [inline src/routes/medias.rs]: is this supposed to be deterministic?

Comment (epsilon-phase) [reply src/routes/medias.rs]: Is it deterministic? It seems to come up with new GUIDs each piece we upload.

Comment (igalic) [reply src/routes/medias.rs]: aah, never mind, i was thinking of something else
(the deterministially random id we use per release for *our* assets)

File: plume-models/src/medias.rs [hunk 2]
```diff
@@ -202,10 +204,9 @@ impl Media {
             .next()
             .map(ToOwned::to_owned)
             .unwrap_or_else(|| String::from("png"));
-        let path =
-            Path::new("static")
-                .join("media")
-                .join(format!("{}.{}", GUID::rand().to_string(), ext));
+        let path = Path::new(&super::CONFIG.media_directory)
+            .join("media")
```
Comment (fdb-hiroshima) [inline plume-models/src/medias.rs]: the `media` part is already included in the default value of media_directory (`static/media`), and is, I think, not required for a custom media directory

Comment (fdb-hiroshima) [inline plume-models/src/medias.rs]: the `media` part is already included in the default value of media_directory (`static/media`). I don't know if it should be removed from here or from there (it depends on how custom directory should appear), but I think there is one too much

Comment (epsilon-phase) [reply plume-models/src/medias.rs]: Oh, thanks for the catch.

**Comment by RAOF** [comment]
I've built a snap from this locally, setting the new `MEDIA_UPLOAD_DIRECTORY` appropriately, and it works!

**Pull request merged**